src/gbfs.js

```diff
@@ -81,12 +81,12 @@ class GBFS {
 
   bikes(withIds = undefined) {
     let { bikes } = this.feedCache[FEED.freeBikeStatus];
-    if (withIds !== undefined) {
-      return bikes.filter((bike) => withIds.includes(bike.bike_id));
```

Comment:
If we do `bike = bikes.filter((bike) => withIds.includes(bike.bike_id))` here, the `calculateVehicleBatteries` will only be run on the filtered amount of vehicles instead of all. This is a performance hit that can be improved.